anyway/infographics_utils.py

```diff
@@ -156,29 +131,31 @@ def serialize(self):
         return output
 
 
-class WidgetCollection:
-    widgets: List[Type[Widget]] = []
+widgets_dict: Dict[str, Type[Widget]] = {}
 
-    def __init__(self):
-        pass
 
-    @staticmethod
-    def get() -> List[Type[Widget]]:
-        return WidgetCollection.widgets
+def get_widget_classes() -> List[Type[Widget]]:
+    return list(widgets_dict.values())
+
+
+def get_widget_class_by_name(name: str) -> Type[Widget]:
+    return widgets_dict[name]
 
-    @staticmethod
-    def register(widget_class: Type[Widget]) -> Type[Widget]:
-        WidgetCollection.widgets.append(widget_class)
-        return widget_class
 
+def register(widget_class: Type[Widget]) -> Type[Widget]:
+    widgets_dict[widget_class.name] = widget_class
+    logging.debug(f"register:{widget_class.name}:{widget_class}\n")
+    return widget_class
 
-@WidgetCollection.register
+
+@register
 class AccidentCountBySeverityWidget(Widget):
-    widget_id: WidgetId = dataclasses.field(init=False, default=WidgetId.accident_count_by_severity)
+    name: str = "accident_count_by_severity"
 
     def __init__(self, request_params: RequestParams):
-        super().__init__(request_params, WidgetId.accident_count_by_severity)
+        super().__init__(request_params, type(self).name)
```

Comment:
Cool. Now this parameter becomes unnecessary. Since `str` is an immutable type (and moreover, the field is not expected to change, i.e. it is `Final`), the class/static variable `name` (initialized in the class body) is a perfectly valid to access as a field - anyone can just access it using `mywidget.name`. As we've discussed, this does not break encapsulation.

```python
class A:
    name = 'unknown'  # here only to get static type checking when we use polymorphism; can also be a property

class B(A):
    name = 'john'

b = B()
print(b.name)  # perfectly legitimate
```

tl;dr: the parameter `name` and the method `get_name()` can be removed. This is not the Java way, but this is perfectly idiomatic Python.

Reply:
Hi @elazarg , I understand your point and thank you for the deep reviews. At the same time I would prefer to leave this PR as it is now, and progress with other tasks. 
Is it OK with you to approved the PR?
Thanks, Ziv

Reply:
Oh sorry, I was not intentionally withholding approval... I was only giving comments. Sorry!